backends/cuda/tracer_cudart_helpers.include.c

```diff
@@ -14,9 +14,10 @@ static void _load_tracer(void) {
       handle = dlopen("libcudart.so", RTLD_LAZY | RTLD_LOCAL);
   if (handle) {
     void* ptr = dlsym(handle, "cudaSetDevice");
-    if (ptr == (void*)&cudaSetDevice) { //opening oneself
-      dlclose(handle);
-      handle = NULL;
+    void* self = dlsym(RTLD_DEFAULT, "cudaSetDevice");
+    if (ptr == self) {  /* opening oneself */
+        dlclose(handle);
+        handle = NULL;
```

Comment:
Have you tried opening yourself by pointing `LTTNG_UST_CUDART_LIBCUDART` to the tracer, and verified it tripped the check?
If you have then I am fine approving it, I agree that it should be good enough. Alternative would be to define a hidden alias before the symbol gets deleted and the use that to check:
```c
static cudaError_t cudaSetDevice_hid(int device) __attribute__ ((alias ("cudaSetDevice")));
```

Reply:
Oh, good point! No I didn't test that, will check now.

Reply:
> Is this PR have 3 fix?
GNU_SOURCE for better parsing, dlfcn for dlsym, and then the RTLD_DEFAULT thingy?

`GNU_SOURCE` is needed to get `RTLD_DEFAULT` extension from `dlfcn.h`.

The include was because I wasn't sure where `dlfcn.h` was coming in (somewhere in the CUDA headers is my guess) and wanted to ensure `GNU_SOURCE` wasn't unset before it was included.

Reply:
Yes, thanks!

Reply:
I assume this:
```
Don't want to open self, closing...
```
Is a printf you've added yourself correct? otherwise I am unsure where it is coming from.

Reply:
Yes, exactly, I should have mentioned that. Also the "`in _load_tracer`".